docs/smtchecker.rst

```diff
@@ -123,6 +126,110 @@ the SMTChecker proves that no overflow is reachable:
     }
 
 
+Assert
+======
+
+An assertion represents an invariant in your code: a property that must be true
+*for all cases*, otherwise there is a bug.
```

Comment:
must be true *at all times* or *in all cases*? But fine with keeping it, though I'd prefer *at all times*.

Reply:
hmmm depends what time means. That property may be broken 2 lines earlier or two lines after, but it's true at that line

Reply:
"for all cases" sounds rather non-idiomatic to me in any case...
"must be true every time the assertion is executed" or "must always be true at the point of the assertion" or...
But feel free to dismiss :-).

Reply:
I feel like that doesn't bring the point I'm trying to make, which is that it's true for all input and storage values

Reply:
Then make it "a property that must be true for all input and storage values" maybe :-)?

Reply:
But yeah, all fine with me.

Reply:
Changed to something different but still related.